psamm/datasource/sbml.py

```diff
@@ -648,6 +649,44 @@ def __init__(self):
         self._namespace = SBML_NS_L3_V1_CORE
         self._sbml_tag = partial(_tag, namespace=self._namespace)
 
+    def _make_safe_id(self, thisid):
+        """Makes sure the ID only has the allowed characters.
+
+        Replaces or deletes the ones that aren't allowed.
+        """
+
+        thisid = thisid.replace('-', '_DASH_')
+        thisid = thisid.replace('/', '_FSLASH_')
+        thisid = thisid.replace('\\', '_BSLASH_')
+        thisid = re.sub(r'\(([a-z])\)$', '_\\1', thisid)
+        thisid = thisid.replace('(', '_LPAREN_')
+        thisid = thisid.replace(')', '_RPAREN_')
+        thisid = thisid.replace('[', '_LSQBKT_')
+        thisid = thisid.replace(']', '_RSQBKT_')
+        thisid = thisid.replace(',', '_COMMA_')
+        thisid = thisid.replace('.', '_PERIOD_')
+        thisid = thisid.replace("'", '_APOS_')
+        thisid = re.sub(r'[^a-zA-Z0-9_]', '', thisid)
+        return thisid
```

Comment:
I would chain the calls to `replace` to avoid all those duplicate `thisid`s, maybe like this:

``` python
id = (id.replace('-', '_DASH_')
        .replace('/', '_FSLASH_')
        .replace('\\', '_BSLASH_')
        ...)
```

You can do the `re.sub()` first and last. Or you could even do something like

``` python
substitutions = {
    '-': '_DASH_',
    '/': '_FSLASH_',
    ...
}
for symbol, escape in iteritems(substitutions):
    id = id.replace(symbol, escape)
```